### PR TITLE
Refactor Gradle files

### PR DESCRIPTION
**Pull request by BenMaruchu**
- Use variables for support library version and make it easy to update the version across all library modules
- update the script to use latest android plugin api

### CODE REVIEW

File: library-core/build.gradle [hunk 1]
```diff
@@ -41,45 +41,45 @@ android {
...
+    implementation 'com.google.android.gms:play-services-maps:11.6.0'
+    implementation('com.mapbox.mapboxsdk:mapbox-android-sdk:5.1.2@aar') { transitive=true }
+    implementation('com.mapbox.mapboxsdk:mapbox-android-services:2.2.0@aar') { transitive=true }
+    implementation 'com.mapbox.mapboxsdk:mapbox-android-plugin-locationlayer:0.1.0'
 
     // images
-    compile 'com.github.bumptech.glide:glide:4.3.1'
-    annotationProcessor 'com.github.bumptech.glide:compiler:4.3.1'
+    implementation 'com.github.bumptech.glide:glide:4.3.1'
+    annotationProcessor 'com.github.bumptech.glide:implementationr:4.3.1'
```
Comment (krtonga) [inline library-core/build.gradle]: Please take a second look at this. I suspect that your find and replace was a bit too inclusive, and that this should still be `compiler:4.3.1`

Comment (BenMaruchu) [reply library-core/build.gradle]: Thanks Kristen I'm fixing it now

**Comment by lykmapipo** [comment]
@krtonga & @BenMaruchu What is the state of this PR?

**Comment by BenMaruchu** [comment]
@lally it can be merged once reviewed

**Review by krtonga — APPROVED** [review]

**Pull request merged**